tutor/src/api/index.coffee

```diff
@@ -0,0 +1,330 @@
+# `TaskActions.loaded` or `TaskActions.FAILED`
+{APIActionAdapter} = require 'shared'
+
+{ makeIdRouteData, makeDefaultRequestData,
+  createActions, readActions, updateActions, deleteActions,
+  actionFrom, createFrom, readFrom, updateFrom, deleteFrom
+} = APIActionAdapter
+
+{setUpAPIHandler} = require './adapter'
+
+{CurrentUserActions} = require '../flux/current-user'
+{CourseActions} = require '../flux/course'
+{CoursePracticeActions} = require '../flux/practice'
+{CourseGuideActions} = require '../flux/guide'
+{JobActions} = require '../flux/job'
+{EcosystemsActions} = require '../flux/ecosystems'
+PerformanceForecast = require '../flux/performance-forecast'
+
+{ScoresActions} = require '../flux/scores'
+{ScoresExportActions} = require '../flux/scores-export'
+{RosterActions, RosterStore} = require '../flux/roster'
+{PeriodActions} = require '../flux/period'
+
+{TaskActions} = require '../flux/task'
+{TaskPanelActions} = require '../flux/task-panel'
+{TaskStepActions} = require '../flux/task-step'
+{StudentIdActions} = require '../flux/student-id'
+{TaskPlanActions, TaskPlanStore} = require '../flux/task-plan'
+{TaskTeacherReviewActions} = require '../flux/task-teacher-review'
+{TaskPlanStatsActions} = require '../flux/task-plan-stats'
+
+{TocActions} = require '../flux/toc'
+{ExerciseActions, ExerciseStore} = require '../flux/exercise'
+{TeacherTaskPlanActions} = require '../flux/teacher-task-plan'
+{StudentDashboardActions} = require '../flux/student-dashboard'
+{CourseListingActions} = require '../flux/course-listing'
+{CCDashboardActions} = require '../flux/cc-dashboard'
+
+{ReferenceBookActions} = require '../flux/reference-book'
+{ReferenceBookPageActions} = require '../flux/reference-book-page'
+{ReferenceBookExerciseActions} = require '../flux/reference-book-exercise'
+{NewCourseActions, NewCourseStore} = require '../flux/new-course'
+{NotificationActions} = require '../flux/notifications'
+
+BOOTSTRAPED_STORES = {
+  user:   CurrentUserActions.loaded
+  courses: CourseListingActions.loaded
+}
+
+startAPI = ->
+  {updateRequestHandlers, connectTrigger, connectToAPIHandler, connectAsAction} = setUpAPIHandler()
+
+  connectAsAction('read', TaskActions, 'task')
+  connectAsAction('delete', TaskActions, 'task')
+  connectAsAction('read', TaskPlanActions, 'task-plan')
+  connectAsAction('delete', TaskPlanActions, 'task-plan')
+
+  updateRequestHandlers(TaskPlanActions, onSuccess: 'saved', createFrom('task-plan'))
+  updateRequestHandlers(TaskPlanActions, onSuccess: 'saved', updateFrom('task-plan'))
+  connectTrigger(TaskPlanActions, trigger: 'save', (id, courseId) ->
+    subject = 'task-plan'
+    requestData = TaskPlanStore.getChanged(id)
+    routeData = {id, courseId}
+
+    requestInfo = if TaskPlanStore.isNew(id)
+      subject: subject
+      topic: courseId
+      action: 'create'
+    else
+      subject: subject
+      topic: id
+      action: 'update'
+
+    {requestInfo, routeData, requestData}
+  )
+
```

Comment:
Some of these seem to get really verbose.  I think we could probably come up with helpers to slim them down a bit,  however I think we should merge this and ponder the issue a bit while we get used to the new loading api.